rdrf/rdrf/views/dashboard_view.py

```diff
@@ -0,0 +1,253 @@
+
+        if context_form_group.is_fixed:
+            return self.patient.default_context(self.registry)
+
+        return None
+
+    def _get_form_link(self, context_form_group, registry_form, context=None):
+        if not context:
+            context = self._get_patient_context(context_form_group)
+
+        if context:
+            return registry_form.get_link(self.patient, context)
+
+        if context_form_group.is_multiple:
+            link, title = context_form_group.get_add_action(self.patient)
+            return link
+
+    def _patient_consent_summary(self):
+        registry_consent_questions = ConsentQuestion.objects.filter(section__registry=self.registry)
+        patient_consents = ConsentValue.objects.filter(patient=self.patient,
+                                                       consent_question__section__registry=self.registry)
+
+        return {
+            'valid': consent_status_for_patient(self.registry.code, self.patient),
+            'completed': patient_consents.count(),
+            'total': registry_consent_questions.count()
+        }
+
+    def _get_module_progress(self):
+        if not self._request.user.has_perm('patients.can_see_data_modules'):
+            return None
+
+        form_progress = FormProgress(self.registry)
+
+        modules_progress = defaultdict(dict)  # {'fixed': {}, 'multi': {}}
+
+        for cfg, context in self._contexts.items():
+            forms_progress = {}
+            key = None
+            for form in cfg.forms:
+                if not (self._request.user.can_view(form)):
+                    continue
+
+                progress_dict = {}
+
+                if cfg.is_fixed:
+                    if not form.has_progress_indicator:
+                        continue
+
+                    key = 'fixed'
+                    progress_dict['link'] = self._get_form_link(cfg, form, context=context)
+                    progress_dict['progress'] = form_progress.get_form_progress(form, self.patient, context)
+                elif cfg.is_multiple:
+                    key = 'multi'
+                    last_completed = None
+
+                    if context:
+                        form_timestamp = self.patient.get_form_timestamp(form, context)
+                        if form_timestamp:
+                            last_completed = date_format(parse_datetime(self.patient.get_form_timestamp(form, context)))
+
+                    progress_dict['link'] = cfg.get_add_action(self.patient)[0]
+                    progress_dict['last_completed'] = last_completed or None
+
+                forms_progress.update({form: progress_dict})
+            if key:
+                modules_progress[key].update({cfg: forms_progress})
+
+        return modules_progress
+
+    def _get_cde_data(self, cfg, form, section, cde):
+        context = self._get_patient_context(cfg)
+
+        if not context:
+            return None
+
+        data = get_clinical_data.__wrapped__(self.registry.code,
```

Comment:
Should probably not rely on this method if it's part of the deprecated reports. Could move it out if it's useful.